tools/go-cli/build.gradle

```diff
@@ -78,3 +44,34 @@ def run(cmd, ignoreError = false) {
         println("Command '${cmd.join(" ")}' failed with exitCode ${proc.exitValue()}")
     }
 }
+
+task dump_os() << {
+    println "os.name = "+getOsName()
+    println "os.arch = "+getOsArch()
+    println "go.name = "+mapOsNameToGoName(getOsName())
+    println "go.arch = "+mapOsArchToGoArch(getOsArch())
+}
+
+def getOsName() {
+    return System.properties['os.name']
+}
+
+def getOsArch() {
+    return System.properties['os.arch']
+}
+
+def mapOsNameToGoName(String osname) {
+    String osname_l = osname.toLowerCase()
+    if (osname_l.contains("nux") || osname.contains("nix")) return "linux"
+    if (osname_l.contains("mac")) return "mac"
+    if (osname_l.contains("windows")) return "windows"
+    return osname_l
+}
+
+def mapOsArchToGoArch(String osarch) {
+    String osarch_l = osarch.toLowerCase()
+    if (osarch_l == "amd64") return "amd64"
+    if (osarch_l.contains("x86_64")) return "amd64"
+    if (osarch_l.contains("i386")) return "386"
```

Comment:
Sure.  Can add that.  FYI.  This logic in this build.gradle could be short lived as it's probably better to write some sort of run-time wsk wrapper that launches the appropriate wsk binary on-the-fly.  Only for openwhisk development environment though (i.e. someone building openwhisk in vagrant and want to easily run the wsk CLI from vagrant and the host system), not for CLI distribution.